src/panel.js

```diff
@@ -0,0 +1,107 @@
+const {Emitter} = require('event-kit')
+
+// Extended: A container representing a panel on the edges of the editor window.
+// You should not create a `Panel` directly, instead use {Workspace::addTopPanel}
+// and friends to add panels.
+//
+// Examples: [status-bar](https://github.com/atom/status-bar)
+// and [find-and-replace](https://github.com/atom/find-and-replace) both use
+// panels.
+module.exports =
+class Panel {
+  /*
+  Section: Construction and Destruction
+  */
+
+  constructor ({item, visible, priority, className}, viewRegistry) {
+    this.item = item
+    this.visible = visible
+    this.priority = priority
+    this.className = className
+    this.viewRegistry = viewRegistry
+    this.emitter = new Emitter()
+    if (this.visible == null) this.visible = true
+    if (this.priority == null) this.priority = 100
+  }
+
+  // Public: Destroy and remove this panel from the UI.
+  destroy () {
+    this.hide()
+    if (this.element) this.element.remove()
+    this.emitter.emit('did-destroy', this)
+    return this.emitter.dispose()
+  }
+
+  getElement () {
+    if (!this.element) {
+      this.element = document.createElement('atom-panel')
+      if (!this.visible) this.element.style.display = 'none'
+      if (this.className) this.element.classList.add(...this.className.split(' '))
```

Comment:
Wouldn't this be cleaner?

~~~js
this.element.classList.add(...this.classList);
~~~